libethereum/Precompiled.cpp

```diff
@@ -1092,7 +1093,228 @@ ETH_REGISTER_PRECOMPILED( getBlockRandom )( bytesConstRef, const PrecompiledCall
+            return { false, toBigEndian( dev::u256( 6 ) ) };
+
+        dev::bytes walletAndSigBytes =
+            _in.cropped( walletAndSigOffset.convert_to< size_t >() + 32, 116 ).toBytes();
+
+        // Extract wallet address (first 20 bytes)
+        dev::Address walletAddress(
+            dev::bytes( walletAndSigBytes.begin(), walletAndSigBytes.begin() + 20 ) );
+        if ( walletAddress == dev::ZeroAddress )
+            return { false, toBigEndian( dev::u256( 7 ) ) };
+        // verify account is not active
+        if ( g_skaleHost->client().countAt( walletAddress ) > 0 )
+            return { false, toBigEndian( dev::u256( 8 ) ) };
+
+        // Parse signature from remaining bytes: r(32), s(32), v(32)
+        dev::h256 r( dev::bytes( walletAndSigBytes.begin() + 20, walletAndSigBytes.begin() + 52 ) );
+        dev::h256 s( dev::bytes( walletAndSigBytes.begin() + 52, walletAndSigBytes.begin() + 84 ) );
+        dev::h256 vBytes(
+            dev::bytes( walletAndSigBytes.begin() + 84, walletAndSigBytes.begin() + 116 ) );
+        _byte_ v = dev::h256::Arith( vBytes ).convert_to< _byte_ >();
+
+        SignatureStruct signature( r, s, v );
+        if ( !signature.isValid() )
+            return { false, toBigEndian( dev::u256( 9 ) ) };
+
+        // Extract transaction data at the offset (has length prefix)
+        if ( _in.size() < dataOffset.convert_to< size_t >() + 32 )
+            return { false, toBigEndian( dev::u256( 10 ) ) };
+
+        bigint const dataLength( parseBigEndianRightPadded( _in, dataOffset, 32 ) );
+        if ( _in.size() <
+             dataOffset.convert_to< size_t >() + 32 + dataLength.convert_to< size_t >() )
+            return { false, toBigEndian( dev::u256( 11 ) ) };
+
+        dev::bytes txnData =
+            _in.cropped( dataOffset.convert_to< size_t >() + 32, dataLength.convert_to< size_t >() )
+                .toBytes();
+        if ( txnData.empty() )
+            return { false, toBigEndian( dev::u256( 12 ) ) };
+
+        // Convert ABI-encoded data to RLP
+        dev::bytes rlpEncodedData;
+        try {
+            rlpEncodedData = abiEncodedArraysToRlp( txnData );
+        } catch ( std::exception& ex ) {
+            std::string strError = ex.what();
+            if ( strError.empty() )
+                strError = "exception without description";
+            BOOST_LOG( getLogger( VerbosityError ) )
+                << "Exception in precompiled/submitCTX/abiEncodedArraysToRlp(): " << strError
+                << "\n";
+            return { false, toBigEndian( dev::u256( 13 ) ) };
+        } catch ( ... ) {
+            BOOST_LOG( getLogger( VerbosityError ) )
+                << "Unknown exception in precompiled/submitCTX/abiEncodedArraysToRlp()\n";
+            return { false, toBigEndian( dev::u256( 14 ) ) };
+        }
+
+        // Get gas price
+        dev::u256 gasPrice = g_skaleHost->getGasPrice();
```

Comment:
How will it work with historic calls? Suggest to pass the block from PrecompiledCallContext